app/controllers/times_controller.rb

```diff
@@ -49,12 +43,24 @@ def validate_times(date, is_date)
     true
   end
 
+  def date_time_exists?(datetime, is_date)
+    query = @project.project_times.where(date_time: datetime, is_date: is_date)
```

Comment:
Nice job changing so many files and code that you were not especially familiar with.